kong/db/declarative/init.lua

```diff
@@ -754,21 +754,14 @@ end
 
 
 function declarative.load_into_cache_with_events(entities, meta, hash)
-
-  -- ensure any previous update finished (we're flipped to the latest page)
-  local ok, err = kong.worker_events.poll()
-  if not ok then
-    return nil, err
-  end
-
```

Comment:
I believe this block may still be necessary, because the `purge` operation inside `load_into_cache` happens before the worker event is triggered.

With this block added back, then this PR is good to go for 2.1.0-rc1!

Reply:
Of course, we can make further progress later, but this PR is a definite improvement.